test_delete_timeline_post_rm_failure is flaky

```
/github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/allure_commons/_allure.py:221: in __call__
    return self._fixture_function(*args, **kwargs)
test_runner/fixtures/neon_fixtures.py:1126: in neon_env_builder
    yield builder
test_runner/fixtures/neon_fixtures.py:854: in __exit__
    self.env.pageserver.assert_no_errors()
test_runner/fixtures/neon_fixtures.py:1704: in assert_no_errors
    assert not errors
E   AssertionError: assert not ['2023-06-11T19:05:08.104732Z ERROR compaction_loop{tenant_id=dc65f23bcf356adb0f9306fe282c90f7}:compact_timeline{timeline=45201520c8d11a51859f1f8922e2d712}: could not compact, repartitioning keyspace failed: Failed to load delta layer /tmp/test_output/test_delete_timeline_post_rm_failure[debug-pg15-None]/repo/tenants/dc65f23bcf356adb0f9306fe282c90f7/timelines/45201520c8d11a51859f1f8922e2d712/000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__000000000149F078-000000000149F0F1\n']
```

https://neon-github-public-dev.s3.amazonaws.com/reports/main/5237007033/index.html#suites/f588e0a787c49e67b29490359c589fae/b85344caa78d5994

--- a/pageserver/src/tenant.rs
+++ b/pageserver/src/tenant.rs
@@ -1461,7 +1461,13 @@ impl Tenant {
             let timelines = self.timelines.lock().unwrap();
             let timelines_to_compact = timelines
                 .iter()
-                .map(|(timeline_id, timeline)| (*timeline_id, timeline.clone()))
+                .filter_map(|(timeline_id, timeline)| {
+                    if timeline.is_active() {
+                        Some((*timeline_id, timeline.clone()))
+                    } else {
+                        None
+                    }
+                })
                 .collect::<Vec<_>>();
             drop(timelines);
             timelines_to_compact

